Lua 5.5 `global` keyword missing highlighting and breaks formatting

### Summary

## Describe the bug

The new Lua 5.5 `global` keyword is not highlighted in Helix. Moreover, using `global` causes the `lua-language-server` integration to fail formatting — the editor does not format the code when this keyword is present.

## Environment

* Helix version: build from master (b74e2694124b9d269d05f6aace42d4756c3bb724)
* OS: Debian 13 Trixie
* Lua-language-server version: **3.16.3**

### Reproduction Steps

<!-- Ideally provide a key sequence and/or asciinema.org recording. --> 

I tried this:

1. Write a Lua 5.5 file using the `global` keyword:

```lua
#!/usr/bin/env lua

global print, os, table, ipairs

local function main()
  local t = table.create(10, 0)
  for k, v in ipairs(t) do
    print(k, v)
  end
  print("Hello, World")
  return 0
end

os.exit(main(), true)
```

I expected this to happen:
- `global` get highlighted
- code get formatted on save / or when the format command is invoked

Instead, this happened:
- `global` is not highlighted
- code does not get formated


### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```
2025-12-24T19:03:01.675 helix_lsp::client [WARN] language server failed to terminate gracefully - failed to parse: invalid type: boolean `true`, expected unit
2025-12-24T19:03:01.678 helix_lsp::client [WARN] language server failed to terminate gracefully - server closed the stream
2025-12-24T19:03:01.678 helix_lsp::transport [ERROR] markdown-oxide err: <- StreamClosed
2025-12-24T19:03:01.678 helix_lsp::transport [ERROR] markdown-oxide err: <- IO(Os { code: 32, kind: BrokenPipe, message: "Broken pipe" })
2025-12-24T19:03:01.680 helix_lsp::transport [ERROR] err: <- Other(failed to send a message to server

Caused by:
    channel closed)
2025-12-24T19:03:01.680 helix_lsp::transport [ERROR] codebook err: <- StreamClosed
2025-12-24T19:03:18.719 helix_lsp::client [WARN] language server failed to terminate gracefully - failed to parse: invalid type: boolean `true`, expected unit

</details>


### Platform

Linux

### Terminal Emulator

 wezterm 20240203-110809-5046fc22

### Installation Method

source

### Helix Version

helix 25.07.1 (b74e2694)

--- a/languages.toml
+++ b/languages.toml
@@ -1522,7 +1522,7 @@ language-servers = [ "lua-language-server" ]
 
 [[grammar]]
 name = "lua"
-source = { git = "https://github.com/tree-sitter-grammars/tree-sitter-lua", rev = "88e446476a1e97a8724dff7a23e2d709855077f2" }
+source = { git = "https://github.com/tree-sitter-grammars/tree-sitter-lua", rev = "e40f5b6e6df9c2d1d6d664ff5d346a75d71ee6b2" }
 
 [[language]]
 name = "luap"

--- a/runtime/queries/lua/highlights.scm
+++ b/runtime/queries/lua/highlights.scm
@@ -53,6 +53,7 @@
 [
  "in"
  "local"
+ "global"
  (break_statement)
  "goto"
 ] @keyword

